hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/MappingMetamodelImpl.java

```diff
@@ -743,6 +743,10 @@ public MappingModelExpressable resolveMappingExpressable(SqmExpressable<?> sqmEx
 			throw new NotYetImplementedFor6Exception( "Resolution of embedded-valued SqmExpressable nodes not yet implemented" );
 		}
 
+		if ( sqmExpressable instanceof EntityTypeImpl ) {
+			throw new NotYetImplementedFor6Exception( "Resolution of entity-valued SqmExpressable nodes not yet implemented" );
+		}
+
 		if ( sqmExpressable instanceof EmbeddableTypeImpl ) {
 			return (MappingModelExpressable) sqmExpressable;
 		}
```

Comment:
This class should be the centralized method to resolve mapping. The above insertion is to ensure regression of the methods annotated with **@NotImplementedYet** in org.hibernate.orm.test.jpa.criteria.components.**ComponentInWhereClauseTest**